File: test/src/cpp_integration/cpp/union_narrowing.cpp
#include "std/tsobject.h"
#include "std/tsnumber.h"

namespace cpp
{
    class UnionTest : public Object
    {
        UnionTest();

        Number *bypass(Number *n) const;
    };

    UnionTest::UnionTest() {}

    Number *UnionTest::bypass(Number *n) const
    {
        return n;
    }
}

File: test/src/compiler/cases/cpp_integration/union_narrowing.ts
import { UnionTest } from "cpp_integration_exts";

{
    const u: number | undefined = 3;

    const c = new UnionTest();

    if (u) {
        const v = c.bypass(u);
        console.assert(v === u, "Pass narrowed union to CXX method");
    } else {
        console.assert(false, "Never");
    }
}
